src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -124,6 +127,65 @@ protected function json($data, $status = 200, $headers = array(), $context = arr
     }
 
     /**
+     * Returns a Response (or BinaryFileResponse) object with original or customized file name and disposition header.
+     *
+     * @param File|string $file        File object (or string with content) to be sent as response
+     * @param null        $fileName    File name to be sent to response or null (will use original file name)
+     * @param string      $disposition Disposition of response (attachment is default, other type is inline)
+     * @param string      $mimeType    Mime type of file will be auto detected on File object (so it can be null) but is mandatory for string
+     *
+     * @return Response|BinaryFileResponse
+     *
+     * @throws \LogicException|\InvalidArgumentException
+     */
+    protected function file($file, $fileName = null, $disposition = ResponseHeaderBag::DISPOSITION_ATTACHMENT, $mimeType = null)
+    {
+        if ($file instanceof File && $file->isReadable()) {
+            $response = new BinaryFileResponse($file);
+            $mimeType = $file->getMimeType();
+        } elseif (is_string($file)) {
+            if ($mimeType === null) {
+                throw new \InvalidArgumentException('You must specify mime type for file from string.');
```

Comment:
@Nicofuma thx, MimeTypeGuesser is now implemented (using BinaryFileResponse class)